app/api/constituency_lookup/[postcode]/[[...address]]/route.ts

```diff
@@ -84,16 +135,54 @@ export async function GET(
     };
     return NextResponse.json(response);
   } else {
-    // TODO: Use DemocracyClub API to lookup postcode and populate the addresses array, so users can select their
-    // specific address, rather than us expecting to know (or find out) their constituency.
     console.log(
       `Multiple constituencies found for postcode ${normalizedPostcode}`,
     );
+
     const response: ConstituencyLookupResponse = {
       postcode: params.postcode,
       addressSlug: params.address?.[0],
       constituencies: constituencies,
     };
+
+    //see if DC api will give us anything useful
+    //TODO might be worth setting a timeout on this?
+    //TODO remove this if statement, (here so we can see DC API failure!)
+    //let dc_data = null;
+    //if (normalizedPostcode != "DE30GU")
+    // dc_data = await fetch_dc_api(normalizedPostcode, params.address?.[0]);
+
+    const dc_data = await fetch_dc_api(normalizedPostcode, params.address?.[0]);
+
+    // See if DC data can return a more useful response.
+    // Possibly it just matches 1 constituency in which case return that
+    // Possibly it gives us an address picker which we can return.
+    if (dc_data?.boundary_changes) {
+      //Democracy Club data shows this postcode has a single constituency
+      const gss =
+        dc_data.boundary_changes.new_constituencies_official_identifier.substring(
+          4,
+        );
+      const dc_constituency = constituencies.filter((constituency) => {
+        return constituency.gss == gss;
+      });
+
+      if (dc_constituency.length === 1) {
+        response.constituencies = dc_constituency;
+      } else {
+        console.log(JSON.stringify(dc_data));
+        console.log(
+          "DC returned boundary change data but it didn't match a single db record??",
+        );
```

Comment:
So in this case, should we lookup the GSS code for the DC constituency in our SQLite DB to see if we can find a match? Looks like we'll just fall through to the `return` at the very end of the function which returns the response based on our database data, but - according to DemoClub - our database data is wrong... :thinking: 

I think there's 2 sides to this:

**For the client/user**: if DC doesn't think their constituency matches _any_ of our possibilities, do we want to show them our possibilities anyway (which might be wrong!). Seems like we could combine our possibilities with the one DC is returning - or we could just choose to trust DC (assuming we can find the GSS code for the constituency in our database...)

**For us**: we should log an error, and maybe have a look into how we can trigger some sort of alert from Vercel to us which we'll pick up, as it implies something pretty dodgy is going on with our data :thinking:

Reply:
So currently this is going to trigger in any Scottish or NI constituency where we hit Democracy Club cos they've got bugs / not got that data yet... 

I'm not sure it's worth putting coding effort into this given it hopefully never happens. But you're right we do ideally want to log something. I'll leave a TODO to this effect?